### PR TITLE
Coalesce module lookups in createRamClass

### PR DESCRIPTION
**Pull request by tajila**
Coalesce module lookups in createRamClass

Combine module lookups in createRamClass.c. Move up the lookup for
module with findModuleForPackage() earlier in the code so that the
module info can be used to do access checking for superclasses and
interfaces (this will come in another PR).

Signed-off-by: tajila <atobia@ca.ibm.com>

Reviewer: @pshipton

### CODE REVIEW

File: runtime/vm/createramclass.cpp [hunk 1]
```diff
@@ -2826,6 +2777,47 @@ internalCreateRAMClassFromROMClass(J9VMThread *vmThread, J9ClassLoader *classLoa
 	/* To prevent deadlock, release the classTableMutex before loading the classes required for the new class. */
 	omrthread_monitor_exit(javaVM->classTableMutex);
 
+	pkgNameLength = packageNameLength(romClass);
```
Comment (pshipton) [inline runtime/vm/createramclass.cpp]: This should be declared and initialized in the block where it is used.

**Comment by pshipton** [comment]
jenkins test sanity plinux

**Comment by JasonFengJ9** [comment]
A changeset I am working on to address #859 is using `ramclass->module` to determine if package access check for superclasses and interfaces should proceed or not.

**Comment by tajila** [comment]
@JasonFengJ9 We might be working on a similar issue. We are currently failing JCKs where a class is incorrectly loading a superclass.  
  
Here is the existing code:
```
			/* ensure that the superclass is visible */
			if (!isROMClassUnsafe) {
				if ((superclass->romClass->modifiers & J9_JAVA_PUBLIC) != J9_JAVA_PUBLIC) {
					if (packageID != superclass->packageID) {
						Trc_VM_CreateRAMClassFromROMClass_superclassNotVisible(vmThread, superclass, superclass->classLoader, classLoader);
						setCurrentExceptionForBadClass(vmThread, superclassName, J9VMCONSTANTPOOL_JAVALANGILLEGALACCESSERROR);
						return FALSE;
					}
				}
			}
```
To really make sure the class is visible we need to to module access checks. 

This PR is refactoring the code to make it easier to do access checks over there. Is that the change you were intending to make?

**Comment by JasonFengJ9** [comment]
@tajila The issue #859 seems due to unnecessary package access check when resolving the classes loading from same module. To make a complete solution, I have to address the classes being resolved, super classes and interface classes loading. When dealing with super/interface classes, I have similar need for `module` access, so your PR is going to help mine though some rebasing work is required.
It appears the JCK failure you are working on was caused by an incorrect super class loading (i.e., shouldn't allowed but proceed), so the issues we are working on need different PR solution but both need `ramclass->module`. It is nice to see you submit this common framework PR first.

**Comment by tajila** [comment]
@pshipton I'm investigating the errors, will post again once its ready

**Comment by tajila** [comment]
@pshipton the changes are ready for another sanity test

**Comment by pshipton** [comment]
sanity testing was done internal to IBM

**Pull request merged**